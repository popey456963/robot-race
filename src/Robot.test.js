const { NORTH } = require("./Constants")
const { createNewRobot, rawDamageRobot, isRobotDead, setRobotPosition } = require("./Robot")

const robot = createNewRobot('example', { x: 1, y: 1 }, NORTH)

test('create new robot', () => {
    expect(robot).toMatchSnapshot()
})

test('damage robot', () => {
    const testRobot = { ...robot }

    expect(testRobot.damage).toBe(0)

    rawDamageRobot(testRobot, 2)

    expect(testRobot.damage).toBe(2)

    rawDamageRobot(testRobot, 100)

    expect(testRobot.damage).toBe(10)
})

test('is robot dead', () => {
    const testRobot = { ...robot }

    expect(isRobotDead(testRobot)).toBe(false)
    rawDamageRobot(testRobot, 100)
    expect(isRobotDead(testRobot)).toBe(true)
})

test('can robot be moved', () => {
    const testRobot = { ...robot }

    expect(testRobot.position).toMatchSnapshot()
    expect(setRobotPosition(robot, { x: 1, y: 2 })).toMatchSnapshot()
})