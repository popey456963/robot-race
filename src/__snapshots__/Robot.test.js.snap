// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can robot be moved 1`] = `
Object {
  "x": 1,
  "y": 1,
}
`;

exports[`can robot be moved 2`] = `undefined`;

exports[`create new robot 1`] = `
Object {
  "checkpoint": Object {
    "x": 1,
    "y": 1,
  },
  "damage": 0,
  "direction": "NORTH",
  "flags": Array [],
  "lives": 3,
  "position": Object {
    "x": 1,
    "y": 1,
  },
  "poweredDown": false,
  "upgrades": Array [],
  "user": "example",
}
`;
